experiment/kind-detect-local-e2e.sh

```diff
@@ -54,12 +54,17 @@ signal_handler() {
 }
 trap signal_handler INT TERM
 
-# build kubernetes / node image, e2e binaries
+# build kubernetes / node image, e2e binaries and ginkgo
 build() {
   # build the node image w/ kubernetes
   kind build node-image -v 1
+  # Ginkgo v1 is used by Kubernetes 1.24 and earlier, fallback if v2 is not available.
+  GINKGO_SRC_DIR="github.com/onsi/ginkgo/v2/ginkgo"
+  if [ ! -d "$GINKGO_SRC_DIR" ]; then
+      GINKGO_SRC_DIR="github.com/onsi/ginkgo/ginkgo"
```

Comment:
My comment was specifically about the v2 path. I'm less sure about this one. It works in current master, but is that also true for older releases?

Reply:
I prefer to err on the side of caution and not switch for v1 ginkgo.

Reply:
> I prefer to err on the side of caution and not switch for v1 ginkgo.

okay, but how about the script is used for testing on older releases? I don't the jobs here, but this could be observed for kind's jobs.https://github.com/kubernetes-sigs/kind/pull/2719

Reply:
That's exactly my point: `vendor/github.com/onsi/ginkgo/ginkgo` was used before this PR. Let's continue to use that and only switch to `github.com/onsi/ginkgo/v2/ginkgo` when needed.